stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -648,27 +648,406 @@ extension Unsafe${Mutable}BufferPointer {
+  ///
+  /// - Precondition: `self.count` >= `source.count`
+  ///
+  /// - Parameter source: A collection of elements to be used to
+  ///   initialize the buffer's storage.
+  /// - Returns: An index to the next uninitialized element in the buffer.
+  @inlinable
+  @_alwaysEmitIntoClient
+  public func initialize(
+    fromContentsOf source: some Collection<Element>
+  ) -> Index {
+    let count = source.withContiguousStorageIfAvailable {
+      guard let sourceAddress = $0.baseAddress, !$0.isEmpty else {
+        return 0
+      }
+      _precondition(
+        $0.count <= self.count,
+        "buffer cannot contain every element from source."
+      )
+      baseAddress?.initialize(from: sourceAddress, count: $0.count)
+      return $0.count
+    }
+    if let count {
+      return startIndex.advanced(by: count)
+    }
+
+    var (iterator, copied) = source._copySequenceContents(initializing: self)
+    _precondition(
+      iterator.next() == nil,
+      "buffer cannot contain every element from source."
+    )
+    return startIndex.advanced(by: copied)
+  }
+
+  /// Updates every element of this buffer's initialized memory.
+  ///
+  /// The buffer’s memory must be initialized or its `Element` type
+  /// must be a trivial type.
+  ///
+  /// - Note: All buffer elements must already be initialized.
+  ///
+  /// - Parameters:
+  ///   - repeatedValue: The value used when updating this pointer's memory.
+  @inlinable // unsafe-performance
+  @_silgen_name("$sSr6assign9repeatingyx_tF")
+  public func update(repeating repeatedValue: Element) {
     guard let dstBase = _position else {
       return
     }
 
-    dstBase.assign(repeating: repeatedValue, count: count)
+    dstBase.update(repeating: repeatedValue, count: count)
   }
-  
+
+  @inlinable
+  @_alwaysEmitIntoClient
+  @available(*, deprecated, renamed: "update(repeating:)")
+  @_silgen_name("_swift_se0370_UnsafeMutableBufferPointer_assign_repeating")
+  public func assign(repeating repeatedValue: Element) {
+    update(repeating: repeatedValue)
+  }
+
+  /// Updates the buffer's initialized memory with the given elements.
+  ///
+  /// The buffer’s memory must be initialized or its `Element` type
+  /// must be a trivial type.
+  ///
+  /// - Parameter source: A sequence of elements to be used to update
+  ///   the buffer's contents.
+  /// - Returns: An iterator to any elements of `source` that didn't fit in the
+  ///   buffer, and the index one past the last updated element in the buffer.
+  @inlinable
+  @_alwaysEmitIntoClient
+  public func update<S: Sequence>(
+    from source: S
+  ) -> (unwritten: S.Iterator, index: Index) where S.Element == Element {
+    var iterator = source.makeIterator()
+    guard var pointer = baseAddress else { return (iterator, startIndex) }
+    for index in indices {
+      guard let element = iterator.next() else {
+        return (iterator, index)
+      }
+      pointer.pointee = element
+      pointer += 1
```

Comment:
Can this pointer be removed?

```suggestion
      self[_unchecked: index] = element
```

Reply:
Yes. I did that in `update(fromContentsOf:)`, but didn't notice this one. Thanks.